libs/designer/src/lib/ui/panel/recommendation/recommendationPanelContext.tsx

```diff
@@ -226,13 +226,14 @@ export const RecommendationPanelContext = (props: CommonPanelProps) => {
         {
           [SELECTION_STATES.AZURE_RESOURCE]: selectedOperation ? <AzureResourceSelection operation={selectedOperation} /> : null,
           [SELECTION_STATES.CUSTOM_SWAGGER]: selectedOperation ? <CustomSwaggerSelection operation={selectedOperation} /> : null,
-          [SELECTION_STATES.DETAILS]: selectedConnector ? (
+          [SELECTION_STATES.DETAILS]: selectedOperationGroupId ? (
             <OperationGroupDetailView
               connector={selectedConnector}
               groupOperations={allOperationsForGroup}
               filters={filters}
               onOperationClick={onOperationClick}
               isLoading={isLoadingOperations || isLoadingOperationGroup}
+              isLoadingConnector={isLoadingConnectors}
```

Comment:
Won't this not let us look at any connectors if any of them are still loading?

Reply:
Yep, I think thats right. I will change to be loading until the `selectedConnector` exists or is different than undefined. Do you think that will work?

Reply:
Yeah I think that would be better, if the connection is slow and they click on a connector, they may sit there waiting a while until all the connectors load

Reply:
Yep, you are right I did a couple of tests with sleeps in other connectors types and this was waiting for all of them to be resolved. Thanks for pointing this out